pkg/sidecar/pod.go

```diff
@@ -64,20 +76,28 @@ func remove(pod corev1.Pod) corev1.Pod {
 		return pod
 	}
 
-	var containers []corev1.Container
-	for _, container := range pod.Spec.Containers {
-		if container.Name != naming.Container() {
-			containers = append(containers, container)
-		}
+	fn := func(c corev1.Container) bool { return c.Name == naming.Container() }
+	pod.Spec.Containers = slices.DeleteFunc(pod.Spec.Containers, fn)
+
+	if featuregate.EnableNativeSidecarContainers.IsEnabled() {
+		// NOTE: we also remove init containers (native sidecars) since k8s 1.28.
+		// This should have no side effects.
+		pod.Spec.InitContainers = slices.DeleteFunc(pod.Spec.InitContainers, fn)
 	}
-	pod.Spec.Containers = containers
 	return pod
 }
 
 // existsIn checks whether a sidecar container exists in the given pod.
 func existsIn(pod corev1.Pod) bool {
-	for _, container := range pod.Spec.Containers {
-		if container.Name == naming.Container() {
+	fn := func(c corev1.Container) bool { return c.Name == naming.Container() }
```

Comment:
this is a nit, but given we use this a few times here, it may be easier to just make it a file scoped var.